frame/staking/src/lib.rs

```diff
@@ -752,6 +753,16 @@ pub trait Config: frame_system::Config + SendTransactionTypes<Call<Self>> {
 	/// is not used.
 	type UnixTime: UnixTime;
 
+	/// This flag is ued to automatically update the validator count per era
+	/// 
+	type EnableAutomaticValidatorUpdatePerEra: Get<bool>;
+	/// Convert current validator count to a new validator count 
+	/// giving the total exposures, min & max validators count, bottom x percent of 
+	/// validators and bottom y percent of validators.
+	type AutomaticValidatorUpdatePerEra: Convert<(u32, Vec<BalanceOf<Self>>, u32, u32, u32, u32), u32>;
+	type BottomXPercentOfValidators: Get<u32>;
+	type BottomYPercentOfValidators: Get<u32>;
```

Comment:
```suggestion
	/// An update function that is executed per era to update the count of validators in an automated way. 
	///
	/// The current count of validators, and stakes of the current elected validators is passed in as the input
	/// to the `Convert` operation. Note that the effect of any update in this function to the validator count will 
	/// be visible in the *next* era. 
	///
	/// If you don't intent do use any automatic updates, simply pass a [`sp_runtime::traits::Identity`].
	type AutomaticValidatorUpdatePerEra: Convert<(u32, Vec<BalanceOf<Self>>), u32>;
```

This is what we are looking for ^^